core/directory.go

```diff
@@ -255,7 +256,10 @@ func (dir *Directory) WithDirectory(ctx context.Context, subdir string, src *Dir
 	}
 
 	st = st.File(llb.Copy(srcSt, srcPayload.Dir, path.Join(destPayload.Dir, subdir), &llb.CopyInfo{
-		CreateDestPath: true,
+		CreateDestPath:      true,
+		CopyDirContentsOnly: true,
+		IncludePatterns:     filter.Include,
+		ExcludePatterns:     filter.Exclude,
```

Comment:
@vito I'm wondering if we should allow setting `AllowWildcard` in `llb.CopyInfo`. Thoughts?

Reply:
I wonder if we should just always set it to `true`. It looks like we did that before. :thinking: I'll make that change and @sipsma can tell me if it's a bad idea. :P

https://github.com/dagger/dagger/blob/c9f7c1148c43f4d2a46c6cacd12269ba1ea9c9e1/core/schema/filesystem.go#L132

Reply:
Sounds good, @vito!

Reply:
Wait, I'm not actually sure how this would be used, since the wildcard is applied to the source directory meaning you'd need to have a `DirectoryID` with a wildcard in it.

You would have to do something like this:

```go
func TestDirectoryWithDirectoryWildcard(t *testing.T) {
	t.Parallel()

	c, ctx := connect(t)
	defer c.Close()

	dir := c.Directory().
		WithNewFile("a.txt").
		WithNewFile("b.txt").
		WithNewFile("c.txt.rar").
		WithNewFile("subdir/d.txt").
		WithNewFile("subdir/e.txt").
		WithNewFile("subdir/f.txt.rar")

	dirID, err := dir.Directory("s*").ID(ctx)
	require.NoError(t, err)

	entries, err := c.Directory().WithDirectory(dirID, "").Entries(ctx)
	require.NoError(t, err)
	require.Equal(t, []string{"d.txt", "e.txt", "f.txt.rar"}, entries)
}
```

The `dir.Directory("s*")` is what feels weird, because that gives you an object that presumes it's going to be used for copying in to somewhere.

Reply:
I'm good with that. The corner case is that it's technically possible for files and directories to have `*` in their name, but like... it's hard to think of a more obscure situation. We can have an optional field to disallow wildcards too to handle that if we want, but defaulting on is good with me.

Reply:
@vito Oh... I think I was misunderstanding how `AllowWildcard` works. `(Include|Exclude)Patterns` already allows wildcards, I'm guessing?

Reply:
> EDIT: I loaded the page before the most recent comment, above was in response to whether it should default on

:laughing: Well, I added it, we can decide whether we want to keep it. It feels a little weird to me since a wildcarded `DirectoryID` won't work for every directory API call, but maybe it'll be useful in actual code.

Reply:
> @vito Ah... I think I was misunderstanding how `AllowWildcard` works. `(Include|Exclude)Patterns` already allows wildcards, I'm guessing?

Yeah, they already support wildcards.

Sounds like this is redundant then, I'll just revert the commit. :)

Reply:
Technically I _may_ need it to **fully** implement the 0.2 API. But that might be something we can just say we won't allow. 

Pull it for now. :shipit: